[BE] feat: 로그인한 사용자 정보 조회 API 기능 및 SecurityConfig 리팩터링

yuyoungrhee opened the pull request:
<!-- PR 제목 예시
ex. [BE] 로그인 API 구현
ex. [AN] 페이지 구현 -->

## 📌 변경 내용 & 이유
<!-- ex. 
- 사용자 로그인 기능 구현
- 로그인 실패 시 에러 메시지 반환 처리 추가 -->
### JwtAuthenticationFilter에서도 허용되는 api들을 검사하지 않도록 수정
- 기존: JwtAuthenticationFilter에서 먼저 api를 검사하는데, token이 null이면 모두 유효하지 않은 토큰 처리가 되어 401 예외 발생.
허용되는 API들도 Jwt Filter에 걸려 401 예외가 발생하는 문제가 있어 수정함
- 현재: 허용되는 api 리스트를 JwtAuthenticationFilter에도 넘겨 사용할 수 있도록 함.

## 📸 스크린샷 (선택)
> UI 변경이 있을 경우 첨부

## 🧪 테스트 방법 (선택)
<!-- 코드 리뷰하는 팀원이 돌려볼 수 있도록 작성해주기 
ex. ReservationTest의 XXXX 테스트메서드로 확인 
ex. 어떤화면의 어떤기능에서 확인 -->
MemberControllerTest, MemberServiceTest 에서 확인 가능

## 📢 논의하고 싶은 내용
<!-- 이러이러해서 이부분을 이러이러하게 구현했는데 이방식이 괜찮은지? -->
없음

## 🧩 관련 이슈
<!-- 이슈 태그: #123 -->
<!-- 이슈 닫기: close #123 -->
close #122 


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **New Features**
  * Introduced comprehensive backend API for Hearit, including endpoints for authentication (local and Kakao login), Hearit management, categories, keywords, bookmarks, and member information.
  * Added admin interfaces and REST APIs for managing Hearits, categories, and keywords with HTML templates and pagination.
  * Implemented JWT-based authentication and authorization, CORS configuration, and OpenAPI (Swagger) documentation.
  * Provided Docker Compose setup for local MySQL databases and development environment configuration.
  * Enabled file resource endpoints for audio and script URLs.

* **Bug Fixes**
  * N/A (initial implementation)

* **Documentation**
  * Added detailed OpenAPI/Swagger documentation for all endpoints.
  * Included admin HTML templates for management UI.

* **Tests**
  * Added integration and unit tests for services and controllers, covering core functionalities and error scenarios.

* **Chores**
  * Updated build configuration, .gitignore, and CI/CD workflows for automated testing and deployment.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

yuyoungrhee committed:
[BE] refactor: JwtAuthenticationFilter에서 허용되는 API에 대해 검사하지 않도록 수정 #122

- 허용되는 API에 대해 Jwt Filter에서도 검사를 중복으로 하게 되는데, 허용하고 다음 필터로 넘겨야되기 떄문.

yuyoungrhee committed:
[BE] feat: Member에 profileImage 필드 추가 #122

- 사용자 프로필 이미지 정보를 저장하고 조회하는 기능에 대응하기 위해
- Member new 생성자 private으로 변경 및 그에 따른 테스트 수정: 현재 정적팩터리메서드 사용 중

yuyoungrhee committed:
[BE] feat: 로그인한 사용자 정보 조회 API 구현 #122

yuyoungrhee committed:
[BE] refactor: admin controller 네이밍 컨벤션 반영 #122

- get -> read로 컨벤션 적용

yuyoungrhee committed:
[BE] Merge branch 'develop-be' into feat/#122-get-member-info #122

# Conflicts:
#	backend/src/main/java/com/onair/hearit/admin/presentation/AdminCategoryController.java
#	backend/src/main/java/com/onair/hearit/admin/presentation/AdminHearitController.java
#	backend/src/main/java/com/onair/hearit/admin/presentation/AdminKeywordController.java

Byesol commented on backend/src/main/java/com/onair/hearit/domain/Member.java:
```diff
@@ -38,26 +38,31 @@ public class Member {
 
     public static Member createLocalUser(String memberId, String nickname, String password) {
-        return new Member(memberId, password, null, nickname);
+        String defaultProfileImage = "http://hearit.o-r.kr/images/default-profile.jpg";
```
이 부분은 상수로 빼거나 properties로 관리하는 게 좋아보여요.

HamBeomJoon commented:
네 리뷰할게요..

Byesol reviewed (commented):
고생하셨습니다! 아래 사항만 확인해주세요

YuyoungRhee commented:
@coderabbitai 해당 PR 범위에 대한 리뷰만 해줘

gabean13 reviewed (commented):
고생하셨어요 멍멍구멍구

yuyoungrhee committed:
[BE] refactor: 기본 이미지 설정을 memberService에서 하도록 변경 #122

- 도메인이 외부 자원에 의존하는 것을 막기 위해 Service로 로직을 뺌
- 기본 프로필 이미지 url을 properties 값으로 빼고, Service에서 @Value로 주입받도록 함

YuyoungRhee commented on backend/src/main/java/com/onair/hearit/application/MemberService.java:
```diff
@@ -0,0 +1,21 @@
+
+    private final MemberRepository memberRepository;
+
+    public MemberInfoResponse getMemberById(Long memberId) {
+        Member member = memberRepository.findById(memberId)
+                .orElseThrow(() -> new NotFoundException("memberId", memberId.toString()));
+        return MemberInfoResponse.from(member);
+    }
```
수정했어용~

yuyoungrhee committed:
[BE] refactor: get 네이밍 컨벤션 반영, byId 제거 #122

yuyoungrhee committed:
[BE] test: 테스트 DisplayName 수정 #122

Byesol reviewed (approved):
리뷰 반영 모두 확인했습니다!!

gabean13 reviewed (approved):
수고하셨어염

JO-eusan reviewed (approved):
수정 사항 확인했습니다.
